clang/test/Sema/builtin-stackaddress.c

```diff
@@ -36,3 +36,8 @@ void* h(unsigned x) {
 // expected-error@+1 {{argument value 1048575 is outside the valid range [0, 65535]}}
 return __builtin_frame_address(0xFFFFF);
 }
+
+void *i() {
+// expected-error@+1 {{too many arguments to function call, expected 0, have 1}}
+return __builtin_stack_address(0);
```

Comment:
```suggestion
  // expected-error@+1 {{too many arguments to function call, expected 0, have 1}}
  return __builtin_stack_address(0);
```
Other test cases I would like to see are:
```
 // As a global variable where there is no stack address to get.
// This should be diagnosed as an error?
void *ptr = __builtin_stack_address();

inline void *what() {
  return __builtin_stack_address();
}

void func() {
  void *ptr = what(); // Is this getting the stack address of? The inline function or is it getting the stack address of the caller? Or depends on optimization level?
}
```
Should we also have tests for what happens with naked functions or other kind of odd situations where the stack may be different from the normal case?

Reply:
> Should we also have tests for what happens with naked functions or other kind of odd situations where the stack may be different from the normal case?

I might need a little bit of help with this, it seems like you can't have non-ASM code inside an `__attribute__((naked))` function, so you can't call `__builtin_stack_address()` inside one.

I'm also not very sure about the odd situations where the stack may be different from the normal case. Are you referring to something similar to an interrupt handler (i.e. `__attribute__((interrupt)) void isr(void*)`?

Using something like `__attribute__((interrupt))` simply translates to an additional function attribute in the generated IR (e.g. `x86_intrcc`). Is it more appropriate to test those scenarios in `llvm/test/CodeGen/${Target}/` instead?